File: RA1/JAVA/EjemploHerencia/Dog.java
package EjemploHerencia;

public class Dog extends Animal{
    private String earType;

    public Dog(String name, String earType){
        super(name);//llamar al constructor de la clase superior, es decir, de clase Animal
        this.earType = earType;
    }

    @Override
    public void makeSound(){
        System.out.println(this.name + " ladra: Guau!!!");
    }

    
    //Polimorfismo: Sobrecarga
    public void run(){
        System.out.println(this.name + " correr");
    }
}

File: RA1/PYTHON/ejemplo_herencia/dog.py
#from <archivo> import <elemento<class>>
from typing import overload
from animal import Animal

class Dog(Animal):
    def __init__(self, name, ear_type):
        #opcion 1: 
        super().__init__(name)
        #opcion 2: Animal.__init__(name)
        self.ear_type = ear_type

    def make_sound(self):
        print(self.name + " ladra: GUAU...")

    def run(self):
        print(self.name + " Corre")

    
    